src/test/regress/sql/gp_tsrf.sql

```diff
@@ -62,3 +62,13 @@ SELECT * FROM t_outer WHERE t_outer.b IN (SELECT generate_series(1, t_outer.b) F
 EXPLAIN (VERBOSE, COSTS OFF)
   SELECT * FROM t_outer WHERE t_outer.b IN (SELECT generate_series(1, t_outer.b)  FROM t_inner);
 DROP TABLE t_outer, t_inner;
+
+-- Check for proper resource deallocation for SRF which has been squelched
+
+-- start_ignore
+DROP TABLE IF EXISTS ao1_srf_test;
+-- end_ignore
+CREATE TABLE ao1_srf_test (a int primary key) WITH (appendonly=true);
+INSERT INTO ao1_srf_test VALUES (1),(2),(3);
+SELECT COUNT (*) FROM (SELECT (gp_toolkit.__gp_aoblkdir('ao1_srf_test'::regclass)).* FROM gp_dist_random('gp_id') LIMIT 1) as onerow;
```

Comment:
Query with a limit and returns random line here, resulting in flaky test, failing in ~30% of runs. But in fact we need to validate that one row is fetched and there were no errors and warnings, exact line content doesn't matter here.
So there were 2 options - to persuade comparer with -- matchsubs and perl regexps, or just rephrase the query with count which still runs same query but return stable number of rows. 
Count looks much simpler.

Reply:
How about such test in other file (gp_toolkit)?

Reply:
Similar test in gp_toolkit did not show the problem with the stability and could run successfully all runs. I think it's because number of rows is different.

Reply:
how about to make here one row too and remove subquery and count (like in gp_toolkit)?

Reply:
I believe that having more complex dataset here will make test coverage wider.

Reply:
I don’t really like that the same test is in different files, and even slightly different.